### PR TITLE
[9] Compatible targets notifications

### PR DESCRIPTION
**Pull request by juanmanuelramallo**
- It broadcasts to all compatible users after a target is created.

### CODE REVIEW

File: app/channels/compatible_targets_channel.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class CompatibleTargetsChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from current_user
```
Comment (GuilleLeopold) [inline app/channels/compatible_targets_channel.rb]: Don't you need to use `stream_for` here?

https://stackoverflow.com/questions/39002150/what-is-the-difference-between-stream-from-and-stream-for-in-actioncable

File: app/jobs/broadcast_compatible_targets_job.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
```
Comment (GuilleLeopold) [inline app/jobs/broadcast_compatible_targets_job.rb]: Why this? You are not handling strings here

File: app/models/target.rb [hunk 3]
```diff
@@ -10,14 +10,24 @@ class Target < ApplicationRecord
...
+  after_create :broadcast_to_compatible_users
+
   def compatible_targets
     Target.where(topic_id: topic_id)
           .where.not(user: user)
           .merge(Target.within(area_length, origin: self))
   end
 
+  def compatible_users
+    User.where id: compatible_targets.pluck(:user_id)
```
Comment (GuilleLeopold) [inline app/models/target.rb]: Use parenthesis here. You can check here our criteria to know when to user parenthesis or not

https://github.com/rootstrap/tech-guides/tree/master/ruby#method-invocation-parens

File: app/models/target.rb [hunk 4]
```diff
@@ -10,14 +10,24 @@ class Target < ApplicationRecord
...
+  end
+
   private
 
+  def broadcast_to_compatible_users
+    BroadcastCompatibleTargetsJob.perform_later self
```
Comment (GuilleLeopold) [inline app/models/target.rb]: Same here

File: spec/jobs/broadcast_compatible_targets_job_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    let(:compatible_target) do
+      create :target, topic: topic, area_length: 2000, lat: lat + 0.001,
+                      lng: lng + 0.001
+    end
+
+    before do
+      compatible_target
+    end
+
+    it 'should broadcast to the users with compatible targets' do
```
Comment (GuilleLeopold) [inline spec/jobs/broadcast_compatible_targets_job_spec.rb]: Don't use `should` in expectation description. 
`it 'broadcasts to the users with compatible targets'`

File: app/channels/compatible_targets_channel.rb (hunk 1 above)
Comment (juanmanuelramallo) [reply app/channels/compatible_targets_channel.rb]: I'll have a look.

File: app/jobs/broadcast_compatible_targets_job.rb (hunk 2 above)
Comment (juanmanuelramallo) [reply app/jobs/broadcast_compatible_targets_job.rb]: Rubocop automatically adds it.

File: app/models/target.rb (hunk 3 above)
Comment (juanmanuelramallo) [reply app/models/target.rb]: 👌

File: app/channels/compatible_targets_channel.rb (hunk 1 above)
Comment (juanmanuelramallo) [reply app/channels/compatible_targets_channel.rb]: You're right https://guides.rubyonrails.org/action_cable_overview.html#streams

File: bin/start [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+foreman start -f ./Procfile.dev
```
Comment (Alxzu) [inline bin/start]: Awesome! 👏

**Pull request merged**